napari/_vispy/canvas.py

```diff
@@ -70,26 +202,398 @@ def _set_theme_change(self, theme: str):
+        (any in vispy.color.get_color_names) or hex starting with # or a tuple | np.array ({3,4},)
+        with values between 0 and 1."""
+        return self._scene_canvas.bgcolor.hex
 
     @bgcolor.setter
-    def bgcolor(self, value):
-        _value = self._background_color_override or value
-        SceneCanvas.bgcolor.fset(self, _value)
+    def bgcolor(self, value: Union[str, npt.ArrayLike]) -> None:
+        self._scene_canvas.bgcolor = value
 
     @property
-    def central_widget(self):
+    def central_widget(self) -> Widget:
         """Overrides SceneCanvas.central_widget to make border_width=0"""
-        if self._central_widget is None:
-            self._central_widget = Widget(
-                size=self.size, parent=self.scene, border_width=0
+        if self._scene_canvas._central_widget is None:
+            self._scene_canvas._central_widget = Widget(
+                size=self.size,
+                parent=self._scene_canvas.scene,
+                border_width=0,
             )
-        return self._central_widget
+        return self._scene_canvas._central_widget
 
-    def _process_mouse_event(self, event):
-        """Ignore mouse wheel events which have modifiers."""
-        if event.type == 'mouse_wheel' and len(event.modifiers) > 0:
+    @property
+    def size(self) -> Tuple[int, int]:
+        """Return canvas size as tuple (height, width) or accepts size as tuple (height, width)
+        and sets Vispy SceneCanvas size as (width, height)."""
+        return self._scene_canvas.size[::-1]
+
+    @size.setter
+    def size(self, size: Tuple[int, int]):
+        self._scene_canvas.size = size[::-1]
+
+    @property
+    def cursor(self) -> QCursor:
+        """Cursor associated with native widget"""
+        return self.native.cursor()
+
+    @cursor.setter
+    def cursor(self, q_cursor: Union[QCursor, Qt.CursorShape]):
+        """Setting the cursor of the native widget"""
+        self.native.setCursor(q_cursor)
+
+    def _on_cursor(self) -> None:
+        """Create a QCursor based on the napari cursor settings and set in Vispy."""
+
+        cursor = self.viewer.cursor.style
+        if cursor in {'square', 'circle'}:
+            # Scale size by zoom if needed
+            size = self.viewer.cursor.size
+            if self.viewer.cursor.scaled:
+                size *= self.viewer.camera.zoom
+
+            size = int(size)
+
+            # make sure the square fits within the current canvas
+            if size < 8 or size > (min(*self.size) - 4):
+                self.cursor = QtCursorVisual['cross'].value
+            elif cursor == 'circle':
+                self.cursor = QtCursorVisual.circle(size)
+            else:
+                self.cursor = QtCursorVisual.square(size)
+        elif cursor == 'crosshair':
+            self.cursor = QtCursorVisual.crosshair()
+        else:
+            self.cursor = QtCursorVisual[cursor].value
```

Comment:
```suggestion
            return
        cursor_ = QtCursorVisual[cursor].value
        if callable(cursor_):
            cursor_ = cursor_()
        self.cursor = cursor_
```

Simplification for easier maintainability?

Reply:
yes.. :(

Reply:
I indeed hoped it could be simplified, but size is the culprit for having it this way.